### PR TITLE
Edited init/new_user methods

### PR DESCRIPTION
**Pull request by JayHenry07**
Added user/contact creation to init method, new user insertion to new_user method

### CODE REVIEW

File: db.py [hunk 1]
```diff
@@ -9,12 +9,40 @@ def __init__(self):
     def init(self):
         """initalize database and tables if not created"""
 
-        pass
+        self.db.execute(
+            """CREATE TABLE user (
+                id integer PRIMARY KEY,
```
Comment (VijayStroup) [inline db.py:14]: For all datatypes, make them CAPITAL like how you dealer `PRIMARY KEY` so it's easier to read.

Comment (VijayStroup) [reply db.py:14]: Also, don't forget to add `NOT NULL` as an option.

File: db.py [hunk 2]
```diff
@@ -9,12 +9,40 @@ def __init__(self):
...
 
     def new_user(self, name: str, password: str):
         """add a new user to the database"""
 
-        pass
+        self.db.execute(
+            """INSERT INTO user VALUES (
+                datetime('now'),
+                datetime('now'),
+                name,
```
Comment (VijayStroup) [inline db.py:41]: You need to bind statements to insert the variables that will be passed into this function.
Currently this just inserts `name` and `password` as the name and password columns.

To do this do:
```py
time = str(datetime.uctnow())
self.db.execute('INSERT INTO user VALUES (?, ?, ?, ?)', (time, time, name, password,))

**Review by VijayStroup — CHANGES_REQUESTED** [review]

File: db.py (hunk 2 above)
Comment (JayHenry07) [reply db.py:41]: Is there a preference from using ' as opposed to " which I used? I saw it both ways when looking up documentation

Comment (VijayStroup) [reply db.py:41]: In Python, there is no difference between a single quote vs a double quote.
Typically we use single quotes to reduce noise in the code so its easier to read.

The only thing though is when you are using single and double quotes in the same string, but that is not an issue here so I would go with just single quotes.

File: db.py [hunk 3]
```diff
@@ -9,12 +9,36 @@ def __init__(self):
     def init(self):
         """initalize database and tables if not created"""
 
-        pass
+        self.db.execute(
+            '''CREATE TABLE user (
```
Comment (VijayStroup) [inline db.py:13]: create table if does not exist

File: db.py [hunk 4]
```diff
@@ -9,12 +9,36 @@ def __init__(self):
...
+                id INTEGER PRIMARY KEY,
+                record_created TEXT NOT NULL,
+                last_logged_in TEXT NOT NULL,
+                username TEXT NOT NULL,
+                password TEXT NOT NULL
+            )''')
+
+        self.db.execute(
+            '''CREATE TABLE contact (
```
Comment (VijayStroup) [inline db.py:22]: create table if does not exist

**Review by VijayStroup — CHANGES_REQUESTED** [review]
Can you also delete the `contact.db` file. It was accidently put into the master branch and was not deleted before you initially cloned your project.

File: db.py [hunk 5]
```diff
@@ -9,12 +9,36 @@ def __init__(self):
     def init(self):
         """initalize database and tables if not created"""
 
-        pass
+        self.db.execute(
+            '''CREATE TABLE IF NOT EXISTS user (
+                id INTEGER PRIMARY KEY,
```
Comment (VijayStroup) [inline db.py:14]: add AUTOINCREMENT option

File: db.py [hunk 6]
```diff
@@ -9,12 +9,36 @@ def __init__(self):
...
+                record_created TEXT NOT NULL,
+                last_logged_in TEXT NOT NULL,
+                username TEXT NOT NULL,
+                password TEXT NOT NULL
+            )''')
+
+        self.db.execute(
+            '''CREATE TABLE IF NOT EXISTS contact (
+                id integer PRIMARY KEY,
```
Comment (VijayStroup) [inline db.py:23]: add AUTOINCREMENT option

Comment (VijayStroup) [inline db.py:23]: capitalize INTEGER

File: db.py [hunk 7]
```diff
@@ -9,12 +9,36 @@ def __init__(self):
...
+                first_name TEXT NOT NULL,
+                last_name TEXT NOT NULL,
+                email TEXT NOT NULL,
+                phone TEXT NOT NULL,
+                record_created TEXT NOT NULL,
+                FOREIGN KEY (user_id) REFERENCES user (id) NOT NULL
```
Comment (VijayStroup) [inline db.py:29]: update this call to this:
```py
self.db.execute(
            '''CREATE TABLE IF NOT EXISTS contact (
                id integer PRIMARY KEY,
                first_name TEXT NOT NULL,
                last_name TEXT NOT NULL,
                email TEXT NOT NULL,
                phone TEXT NOT NULL,
                record_created TEXT NOT NULL,
                user_id INTEGER NOT NULL,
                FOREIGN KEY (user_id) REFERENCES user (id)
            )''')
```

**Review by VijayStroup — CHANGES_REQUESTED** [review]

**Comment by VijayStroup** [comment]
Here is are full proof methods for this.
```py
import sqlite3
from datetime import datetime

class DB:
    def __init__(self):
        self.con = sqlite3.connect('contact.db', check_same_thread=False)
        self.db = self.con.cursor()

    def init(self):
        """initalize database and tables if not created"""

        self.db.execute(
            '''CREATE TABLE IF NOT EXISTS user (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                record_created TEXT NOT NULL,
                last_logged_in TEXT NOT NULL,
                username TEXT NOT NULL UNIQUE,
                password TEXT NOT NULL
            )''')

        self.db.execute(
            '''CREATE TABLE IF NOT EXISTS contact (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                first_name TEXT NOT NULL,
                last_name TEXT NOT NULL,
                email TEXT NOT NULL,
                phone TEXT NOT NULL,
                record_created TEXT NOT NULL,
                user_id INTEGER NOT NULL,
                FOREIGN KEY (user_id) REFERENCES user (id)
            )''')
        
        self.con.commit()

    def new_user(self, name: str, password: str) -> tuple:
        """add a new user to the database"""
        
        time = str(datetime.utcnow())

        try:
            self.db.execute('''INSERT INTO user 
                            (record_created, last_logged_in, username, password) 
                            VALUES (?, ?, ?, ?)''',
                            (time, time, name, password))
            self.con.commit()
            return (0, None)
        except sqlite3.Error as e:
            if type(e.__class__) == type(sqlite3.IntegrityError):
                return (1, 'Username already exists')
            else:
                print(e, e.__class__)
                return (2, e)
```

**Review by VijayStroup — CHANGES_REQUESTED** [review]
on line 6 change the connection to this:
`self.con = sqlite3.connect('contact.db', check_same_thread=False)`

**Review by VijayStroup — APPROVED** [review]

**Pull request merged**